WordPress/gutenberg
# Button: Fix tertiary destructive hover style

Found when testing #62480

## What?

Fixes the hover/active colors of a Button with `variant="tertiary"` and `isDestructive`.

## Testing Instructions

See the "Tertiary" story for Button in Storybook and add a `isDestructive` prop.

For regression testing, try `npm run storybook:e2e:dev`. (The changes shouldn't affect other variants though)

## Screenshots or screencast 

### Before
https://github.com/user-attachments/assets/b820dc23-1968-422f-81e6-a5b31da23a3b

### After
https://github.com/user-attachments/assets/508b7cf5-556d-490a-bad4-87133d123e68
